src/material-experimental/mdc-checkbox/checkbox.ts

```diff
@@ -305,6 +302,14 @@ export class MatCheckbox implements AfterViewInit, OnDestroy, ControlValueAccess
     this.checked = !this.checked;
   }
 
+  /**
+   * Marks the component for check. This can be called after inputs have been updated
```

Comment:
Not sure whether we want to add an explicit API for this because we'll probably end up adding it everywhere.

Reply:
This is just a proposal yet :smile: It's better than adding `markForCheck` multiple times to multiple setters IMO

Reply:
I agree with removing it from setters, but IMO we don’t need the extra API either. If people have access to an instance with ViewChild, then they can also set these values through an input.

Reply:
That is a good way to explain why we want to remove `markForCheck` from the inputs. :+1:

There are a few tests for the checkbox that actually assert that it works with `ViewChild`, but we can remove these from the prototype I think. I'll update the proposal as I prefer that as well.